src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -151,6 +151,28 @@ public override async Task RunAsync()
             // Get the WorkerRuntime
             var workerRuntime = GlobalCoreToolsSettings.CurrentWorkerRuntime;
 
+            // Determine the appropriate default targetFramework
+            // NOTE: .NET 7.0 is only supported on dotnet-isolated
+            // TODO: Include proper steps for publishing a .NET Framework 4.8 application
+            if (workerRuntime == WorkerRuntime.dotnetIsolated)
+            {
+                string projectFilePath = ProjectHelpers.FindProjectFile(functionAppRoot);
+                if (projectFilePath != null)
+                {
+                    var projectRoot = ProjectHelpers.GetProject(projectFilePath);
+                    var targetFramework = ProjectHelpers.GetPropertyValue(projectRoot, Constants.TargetFrameworkElementName);
+                    switch (targetFramework)
+                    {
+                        case "net7.0":
+                            _requiredNetFrameworkVersion = "7.0";
+                            break;
```

Comment:
please add the same for `dotnet-isolated|6.0`

Reply:
Updating from an offline conversation: From [Supported languages in Azure Functions | Microsoft Docs](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#languages-by-runtime-version), it doesn't look like .NET 5 is expected to be supported by Functions V4, and since users should be using Core Tools V4 to publish Functions V4 apps, I think it should be fine to just check for .NET 7.0. The default value for `_requiredNetFrameworkVersion` should also be `"6.0"` already.